filebeat/beater/filebeat.go

```diff
@@ -392,6 +397,9 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	adiscover.Stop()
 	crawler.Stop()
 
+	// We stop the manager and destroy his connection to Elastic Agent.
+	b.Manager.Stop()
```

Comment:
Should this be before the `waitPublished` below? When a beat is signaled to stop by the Elastic Agent it reports `STOPPING` back to the Elastic Agent. I think it would be better to always stop the manager after all the events have been published.